### PR TITLE
An example of using pounders for emittance minimization

### PR DESCRIPTION
**Pull request by jmlarson1**

### CODE REVIEW

File: pounders/py/tests/Test_emittance_opt.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    X0 = np.random.uniform(0, 1, (1, n))  # starting parameters for the optimizer
+    nfmax = int(100)  # Max number of evaluations to be used by optimizer
+    Low = -1 * np.ones((1, n))  # 1-by-n Vector of lower bounds
+    Upp = np.ones((1, n))  # 1-by-n Vector of upper bounds
+    printf = True
+
+    # Don't adjust these:
+    hfun = general_h_funs.emittance_h
+    combinemodels = general_h_funs.emittance_combine
+    m = 3
```
Comment (nneveu) [inline pounders/py/tests/Test_emittance_opt.py:31]: Can we get end of line comments here on what m/gtol/delta/nfs are?

File: pounders/py/tests/Test_emittance_opt.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    [Xout, Fout, flag, xkinout] = pounders(call_beamline_simulation, X0, n, mpmax, nfmax, gtol, delta, nfs, m, F0, xind, Low, Upp, printf, 1, hfun, combinemodels)
+
+    assert flag != 1, "pounders crashed"
+
+    evals = Fout.shape[0]
+    h = np.zeros(evals)
+
+    for i in range(evals):
+        h[i] = hfun(Fout[i, :])
+
```
Comment (nneveu) [inline pounders/py/tests/Test_emittance_opt.py:50]: What's being done with this h?

**Review by nneveu — APPROVED** [review]
Thanks for adding me to review!

File: pounders/py/tests/Test_emittance_opt.py (hunk 2 above)
Comment (jmlarson1) [reply pounders/py/tests/Test_emittance_opt.py:50]: This was just to compute the emittances after the optimization was done. (Those values are computed/used inside of the optimization, but they aren't currently being returned.) I'll add a note that this is the case.

File: pounders/py/tests/Test_emittance_opt.py (hunk 1 above)
Comment (jmlarson1) [reply pounders/py/tests/Test_emittance_opt.py]: Great idea. I've added these comments now.

File: pounders/py/tests/Test_emittance_opt.py (hunk 2 above)
Comment (jmlarson1) [reply pounders/py/tests/Test_emittance_opt.py]: (I also now assert that the minimum emittance has been identified.)

**Review by nneveu — APPROVED** [review]
Looks good, thanks for the comments and min emit change!

**Review by mmenickelly — APPROVED** [review]

**Pull request merged**